src/contracts/types.ts

```diff
@@ -0,0 +1,38 @@
+import { EventEmitter } from 'events'
+import { PromiEvent, TransactionConfig, TransactionReceipt, EventLog } from 'web3-core'
+import { EventOptions, SendOptions, EstimateGasOptions } from 'web3-eth-contract'
+
+type CallTxOptions = Pick<TransactionConfig, 'from' | 'gas' | 'gasPrice'>
+
+type CallTxCallback = <T>(error: Error | null, result: T) => void
+type SendTxCallback = (error: Error | null, transactionHash: string) => void
+type EstimateGasTxCallback = (error: Error | null, gasAmount: number) => void
+
+export type Callback<T> = (error: Error, result: T) => void
+
+export interface TransactionObject<T, U extends any[] = []> {
+  arguments: U
+
+  call(callback: CallTxCallback): Promise<T>
```

Comment:
Ah, i see. Disregard my comment about the return values of `constants` above

Reply:
This thing will also change with the new typechain

Reply:
Yep, for now is just refactoring, I didn't want to make the changes yet, so it would be more difficult to review